pkgs/development/compilers/rust/rustc.nix

```
@@ -100,6 +100,9 @@ in stdenv.mkDerivation rec {
     "${setHost}.musl-root=${pkgsBuildHost.targetPackages.stdenv.cc.libc}"
   ] ++ optionals stdenv.targetPlatform.isMusl [
     "${setTarget}.musl-root=${pkgsBuildTarget.targetPackages.stdenv.cc.libc}"
+  ] ++ optionals stdenv.isDarwin [
```

Jemalloc does not build on aarch64-darwin https://github.com/NixOS/nixpkgs/issues/152056

Let's use jemalloc on x86_64-darwin only

```suggestion
  ] ++ optionals (stdenv.isDarwin && stdenv.isx86_64) [
```

---

Updated, thanks!